src/colmap/geometry/pose_prior.h

```diff
@@ -29,59 +29,232 @@
 
 #include <Eigen/Core>
+#include <Eigen/Geometry>
 
 namespace colmap {
 
+// Represents a pose prior defined by a world coordinate system and its relative
+// transformation to the camera.
+//
+// Commonly used in absolute pose estimation and global bundle adjustment,
+// leveraging prior pose information from sources such as GPS, IMU,
+// previously reconstructed camera poses with covariances, or other systems
+// providing pose estimates with uncertainty.
+//
+// Internally, the pose is stored as a rigid transformation (rotation +
+// translation). When SetRotation() is called, the translation is adjusted to
+// preserve the camera center. Likewise, SetPosition() updates the translation
+// without altering the rotation.
+//
+// Special cases:
+// - If the rotation is invalid, the translation is set to -position.
+// - For WGS84 systems, the position is ordered as longitude, latitude, height,
+//   and rotation is meaningless.
+//
+// The caller must check pose validity using HasValid*() before use.
 struct PosePrior {
  public:
+  static const Eigen::Vector3d kInvalidTranslation;
+  static const Eigen::Quaterniond kInvalidRotation;
+  static const Eigen::Matrix3d kInvalidCovariance3x3;
+  static const Eigen::Matrix6d kInvalidCovariance6x6;
+
   MAKE_ENUM_CLASS(CoordinateSystem,
                   -1,
                   UNDEFINED,  // = -1
                   WGS84,      // = 0
                   CARTESIAN   // = 1
   );
 
-  Eigen::Vector3d position =
-      Eigen::Vector3d::Constant(std::numeric_limits<double>::quiet_NaN());
-  Eigen::Matrix3d position_covariance =
-      Eigen::Matrix3d::Constant(std::numeric_limits<double>::quiet_NaN());
   CoordinateSystem coordinate_system = CoordinateSystem::UNDEFINED;
 
+  Rigid3d cam_from_world = Rigid3d(kInvalidRotation, kInvalidTranslation);
+
   PosePrior() = default;
-  explicit PosePrior(const Eigen::Vector3d& position) : position(position) {}
-  PosePrior(const Eigen::Vector3d& position, const CoordinateSystem system)
-      : position(position), coordinate_system(system) {}
-  PosePrior(const Eigen::Vector3d& position, const Eigen::Matrix3d& covariance)
-      : position(position), position_covariance(covariance) {}
+
+  explicit PosePrior(const Eigen::Vector3d& position)
+      : cam_from_world(kInvalidRotation, -position) {}
+
+  PosePrior(CoordinateSystem system, const Eigen::Vector3d& position)
+      : coordinate_system(system),
+        cam_from_world(kInvalidRotation, -position) {}
+
+  PosePrior(const Eigen::Vector3d& position,
+            const Eigen::Matrix3d& position_covariance)
+      : cam_from_world(kInvalidRotation, -position),
+        position_covariance(position_covariance) {}
+
+  PosePrior(CoordinateSystem system,
+            const Eigen::Vector3d& position,
+            const Eigen::Matrix3d& position_covariance)
+      : coordinate_system(system),
+        cam_from_world(kInvalidRotation, -position),
+        position_covariance(position_covariance) {}
```

Comment:
I find it confusing to store the cam_from_world pose but the world_from_cam covariance. What justifies this design decision?